agent/pkg/app/main.go

```diff
@@ -24,36 +25,38 @@ var (
 )
 
 func LoadExtensions() {
-	Extensions = make([]*tapApi.Extension, 4)
+	Extensions = make([]*tapApi.Extension, 0)
 	ExtensionsMap = make(map[string]*tapApi.Extension)
 
-	extensionAmqp := &tapApi.Extension{}
-	dissectorAmqp := amqpExt.NewDissector()
-	dissectorAmqp.Register(extensionAmqp)
-	extensionAmqp.Dissector = dissectorAmqp
-	Extensions[0] = extensionAmqp
-	ExtensionsMap[extensionAmqp.Protocol.Name] = extensionAmqp
-
 	extensionHttp := &tapApi.Extension{}
 	dissectorHttp := httpExt.NewDissector()
 	dissectorHttp.Register(extensionHttp)
 	extensionHttp.Dissector = dissectorHttp
-	Extensions[1] = extensionHttp
+	Extensions = append(Extensions, extensionHttp)
 	ExtensionsMap[extensionHttp.Protocol.Name] = extensionHttp
 
-	extensionKafka := &tapApi.Extension{}
-	dissectorKafka := kafkaExt.NewDissector()
-	dissectorKafka.Register(extensionKafka)
-	extensionKafka.Dissector = dissectorKafka
-	Extensions[2] = extensionKafka
-	ExtensionsMap[extensionKafka.Protocol.Name] = extensionKafka
-
-	extensionRedis := &tapApi.Extension{}
-	dissectorRedis := redisExt.NewDissector()
-	dissectorRedis.Register(extensionRedis)
-	extensionRedis.Dissector = dissectorRedis
-	Extensions[3] = extensionRedis
-	ExtensionsMap[extensionRedis.Protocol.Name] = extensionRedis
+	if dbgctl.MizuTapperDisableExtensions {
```

Comment:
```suggestion
	if !dbgctl.MizuTapperDisableNonHttpExtensions {
```
For after the merge to debug/profile: Even `MizuTapperDisableNonHttpExtensions` is too specific. Maybe have one variable per extension? `MizuTapperDisableHttpExtensione`, `MizuTapperDisableNonAmqpExtension`, etc.